packages/api/internal/handlers/sandbox_create.go

```diff
@@ -89,6 +95,18 @@ func (a *APIStore) PostSandboxesWithoutResponse(c *gin.Context, ctx context.Cont
 		attribute.String("env.firecracker.version", firecrackerVersion),
 	)
 
+	telemetry.ReportEvent(ctx, "waiting for parallel lock")
+	limitErr := postSandboxParallelLimit.Acquire(ctx, 1)
+	if limitErr != nil {
+		errMsg := fmt.Errorf("error when acquiring parallel lock: %w", limitErr)
+		telemetry.ReportCriticalError(ctx, errMsg)
+
+		a.sendAPIStoreError(c, http.StatusInternalServerError, "Too many requests")
```

Comment:
The error message isn't probably true, acquire returns only error when context is cancelled

Reply:
👀 Not sure if the error message is then overriden by the gin handled. Will check this.